eureka/niriss_class.py

```diff
@@ -0,0 +1,516 @@
+        data : np.ndarray
+           3D array of science group images.
+        err : np.ndarray
+           3D array of errors associated with the science
+           frames.
+        dq : np.ndarray
+           3D array of data quality masks, indicating the
+           location of bad pixels.
+        var : np.ndarray
+           3D array of poisson estimated variances.
+        v0 : np.ndarray
+           3D array of rnoise estimated variances.
+        meta : np.ndarray
+           The array of ASDF additional meta data within
+           the FITS file.
+        median : np.ndarray
+           Median frame of all science images.
+        """
+        hdu = fits.open(os.path.join(self.data_dir, self.filename))
+
+        self.mhdr = hdu[0].header        # Sets in the meta data header
+        self.shdr = hdu['SCI',1].header  # Sets the science data header
+
+        self.intend = hdu[0].header['NINTS'] + 0.0
+        self.time   = np.linspace(self.mhdr['EXPSTART'],
+                                  self.mhdr['EXPEND'],
+                                  int(self.intend) )
+
+        self.time_units = 'BJD_TDB'
+        self.inttime  = hdu[0].header['EFFINTTM']
+
+        # Loads all the data in
+        self.data = hdu['SCI',1].data + 0.0
+        self.raw_data = hdu['SCI',1].data + 0.0
+        self.err  = hdu['ERR',1].data + 0.0
+        self.dq   = hdu['DQ', 1].data + 0.0
+
+        self.var  = hdu['VAR_POISSON',1].data * self.inttime**2.0
+        self.v0   = hdu['VAR_RNOISE', 1].data * self.inttime**2.0
+
+        self.meta = hdu[-1].data
+
+        # Removes NaNs from the data & error/variance arrays
+        self.data[np.isnan(self.data)==True] = 0.0
+        self.err[ np.isnan(self.err )==True] = 0.0
+        self.var[ np.isnan(self.var )==True] = 0.0
+        self.v0[  np.isnan(self.v0  )==True] = 0.0
+
+        print(hdu['DQ', 1].data.shape)
+
+        self.median = np.nanmedian(self.data, axis=0)
+        hdu.close()
+
+        return
+
+    def setup_f277(self, filename):
+        """
+        Opens and assigns proper attributes for the F277W
+        filter observations.
+
+        Parameters
+        ----------
+        filename : str
+           The name of the F277W FITS file.
+
+        Attributes
+        ----------
+        f277 : np.ndarray
+           Science images from this filter.
+        """
+        hdu = fits.open(os.path.join(self.data_dir,
+                                     filename) )
+        self.f277 = hdu[1].data + 0.0
+
+        hdu.close()
+        return
+
+
+    def clean_up(self):
+        """
+        The `clean_up` routine interpolated over bad pixel values,
+        which are marked in the data quality images (`self.dq`).
+        This routine removes bad quality pixels from the following
+        images:
+           - `self.data`
+           - `self.err`
+           - `self.var`
+        """
+        print('Cleaning data . . .')
+        self.data = interpolating_image(self.data, mask=self.dq)
```

Comment:
Why is this never called in the s3_reduce_niriss way of running the code?

Reply:
oop because I forgot. added now